src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -5,324 +5,103 @@
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::FP16:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::FP16>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::BF16:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::BF16>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::U64:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::U64>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::I64:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::I64>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::U8:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::U8>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::BOOL:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::BOOL>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        default:
-            IE_THROW() << "Unsupported input precision " << it.second->getTensorDesc().getPrecision();
-        }
-        _inputs[it.first] = res;
+void SyncInferRequest::share_tensors_with_batched_req(const std::set<std::string>& batched_inputs,
+                                                      const std::set<std::string>& batched_outputs) {
+    for (const auto& it : get_inputs()) {
+        ov::SoPtr<ov::ITensor> res;
+        res = create_shared_tensor_on_batched_tensor(m_batched_request_wrapper->_infer_request_batched->get_tensor(it),
```

Comment:
`m_batched_request_wrapper->_infer_request_batched->get_tensor(it)._so` can be `nullptr`, in this case you need to set so from the infer request.

Reply:
@ilyachur  the _so had been add at 
https://github.com/openvinotoolkit/openvino/blob/e8dc0781b852a03bcd9b0b5dacdcd69edb0771df/src/plugins/auto_batch/src/sync_infer_request.cpp#L26-L30
Is this fine?

Reply:
My point is that `batched_tensor._so` can be nullptr. In this case you need to use so from the hardware infer request

Reply:
Got it! Had modified:
https://github.com/openvinotoolkit/openvino/blob/0484335be227db8723cfabb5baa6dc09e42af0a0/src/plugins/auto_batch/src/sync_infer_request.cpp#L53